autoload/beancount.vim

```diff
@@ -56,14 +66,29 @@ function! beancount#complete_account(findstart, base)
         endif
         let b:beancount_accounts = beancount#find_accounts(l:root)
     endif
-    let l:pattern = '\V' . substitute(a:base, ":", '\\S\\*:\\S\\*', "g")
+
+    if g:beancount_account_completion ==? 'chunks'
+        let l:pattern = '\V' . substitute(a:base, ":", '\\[^ \\t:]\\*:', "g") . '\[^ \t:]\*'
```

Comment:
In beancount, accounts cannot have spaces, so we could change the regex to `\\[^:]\\*:`

Also, the final `\[^ \t:]\*` seems unnecessary, since it won't exclude any matches. Unless you mean `\[^:]\*\$`.